### PR TITLE
Module 3. Working with a string as an array of characters

### PR DESCRIPTION
**Pull request by pmiakish**

### CODE REVIEW

File: module_3/src/com/epam/stringasarrayofcharacters/StringAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.epam.stringasarrayofcharacters;
+
+public class StringAnalyzer {
+
+    public static int countAmountOfDigits(String str) {
+        int counterOfDigits = 0;
+        char[] characters = str.toCharArray();
```
Comment (lovelyserafima) [inline module_3/src/com/epam/stringasarrayofcharacters/StringAnalyzer.java:7]: can be final
```suggestion
        final char[] characters = str.toCharArray();
```

File: module_3/src/com/epam/stringasarrayofcharacters/StringAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        for (char ch : characters) {
+            if ((int) ch > 47 && (int) ch < 58) {
```
Comment (lovelyserafima) [inline module_3/src/com/epam/stringasarrayofcharacters/StringAnalyzer.java:9]: name constants what is 47 and 58

File: module_3/src/com/epam/stringasarrayofcharacters/StringAnalyzer.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                counterOfDigits++;
+            }
+        }
+        return counterOfDigits;
+    }
+
+    public static int countAmountOfNumbers(String str) {
+        String strWithoutSeparators = str.replace(".", "").replace(",", "");
```
Comment (lovelyserafima) [inline module_3/src/com/epam/stringasarrayofcharacters/StringAnalyzer.java:17]: it is a cheat, unfortunately you can't use it yet
```suggestion

```

File: module_3/src/com/epam/stringasarrayofcharacters/StringModifier.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public static String removeRedundantSpaces(String str) {
+        StringBuilder stringWithoutRedundantSpaces = new StringBuilder();
+        char[] characters = str.toCharArray();
+        stringWithoutRedundantSpaces.append(characters[0]);
+        for (int i = 1; i < characters.length; i++) {
+            if (!(characters[i] == ' ' && characters[i - 1] == ' ')) {
+                stringWithoutRedundantSpaces.append(characters[i]);
+            }
+        }
+        return stringWithoutRedundantSpaces.toString().trim();
```
Comment (lovelyserafima) [inline module_3/src/com/epam/stringasarrayofcharacters/StringModifier.java:34]: good implementation but don't use StringBuilder

File: module_3/src/com/epam/stringasarrayofcharacters/StringModifier.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    // task 02
+    public static String changeSubstringInString(String str) {
+        final int wordLength = 4;
+        final char[] charactersOfStr = str.toCharArray();
+        String editedStr = "";
+        for (int i = 0; i <= charactersOfStr.length - wordLength; i++) {
+            if (charactersOfStr[i] == 'w' && charactersOfStr[i + 1] == 'o' && charactersOfStr[i + 2] == 'r' &&
+                    charactersOfStr[i + 3] == 'd') {
+                editedStr = editedStr.concat("letter");
+                i += 3;
```
Comment (lovelyserafima) [inline module_3/src/com/epam/stringasarrayofcharacters/StringModifier.java:32]: why 3?

Comment (pmiakish) [reply module_3/src/com/epam/stringasarrayofcharacters/StringModifier.java:32]: to check the next character after occurrence of 'word' (taking into account the increment of 'i' during the next iteration)

**Pull request merged**